File: PillarsDivTwo/solve/python/PillarsDivTwoSolver.py
#! /usr/bin/python3

import os, sys
import PillarsDivTwo

def init():
    default_path = os.path.join(os.getenv("HOME"), ".gettc")
    gettc_home = os.path.abspath(os.getenv("GETTC_HOME", default_path))
    include_dir = os.path.join(gettc_home, "include/python")
    sys.path.append(include_dir)

def main():
    import topcoder as tc
    with open(sys.argv[1], "r") as fi:
        input = fi.read()
        reader = tc.Reader(input)

        height = reader.next("int[]")
        reader.next()
        w = reader.next("int")

    result = PillarsDivTwo.maximalLength(height, w)
    with open(sys.argv[2], "w") as fo:
        fo.write(tc.write(result, "double"))

if __name__ == "__main__":
    init()
    main()

File: PillarsDivTwo/solve/javascript/PillarsDivTwoSolver.js
#! /usr/bin/env node

var fs = require("fs");
var path = require("path");
var tc = null;

function init() {
  var gettcHome = process.env.GETTC_HOME || path.join(process.env.HOME, ".gettc");
  var includeDir = path.join(gettcHome, "include", "javascript");
  tc = require(path.join(includeDir, "topcoder"));
}

function main() {
  try {
    var input = fs.readFileSync(process.argv[2], { encoding: "ascii" });
    var reader = new tc.Reader(input);
        var height = reader.next('int[]'); reader.next();
    var w = reader.next('int');

    var PillarsDivTwo = require("./PillarsDivTwo");
    var result = PillarsDivTwo.maximalLength(height, w);

    var writer = new tc.Writer();
    writer.next(result, "double");
    fs.writeFileSync(process.argv[3], writer.toString(), { encoding: "ascii" });
  } catch (err) {
    console.log(err.toString());
    console.trace();
  }
}

init();
main();
